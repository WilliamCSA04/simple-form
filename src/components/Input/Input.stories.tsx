import React, { useState } from 'react';
import Input from '.';

export default {
  title: 'Input',
  parameters: {
    info: { inline: true },
  },
};

export const Default = () => {
  const [text, setText] = useState('');
  return <Input value={text} onChange={(e) => setText(e.target.value)} />;
};
