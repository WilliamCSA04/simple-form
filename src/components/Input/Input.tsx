import * as React from 'react';

export interface Props {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  value: string;
  id?: string;
}

export default function Input({ id, onChange, value }: Props) {
  return <input id={id} onChange={onChange} value={value} />;
}
