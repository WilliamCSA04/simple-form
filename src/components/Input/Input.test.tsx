import React from 'react';
import { render } from '@testing-library/react';
import Input from '.';

describe('<Input /> snapshot', () => {
  it('renders the component with default props', () => {
    const container = render(<Input id="Test" value="" onChange={() => {}} />);
    expect(container.asFragment()).toMatchSnapshot();
  });
});
