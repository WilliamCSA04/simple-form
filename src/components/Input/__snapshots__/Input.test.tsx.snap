// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> snapshot renders the component with default props 1`] = `
<DocumentFragment>
  <input
    id="Test"
    value=""
  />
</DocumentFragment>
`;
